runtime/ts/keymgmt/webcrypto.ts

```diff
@@ -0,0 +1,276 @@
+        return crypto.subtle.decrypt({
+            name: this.algorithm(),
+            iv: this.iv,
+        }, this.sessionKey, buffer);
+    }
+
+    encrypt(buffer: ArrayBuffer): PromiseLike<ArrayBuffer> {
+        return crypto.subtle.encrypt(
+            {
+                name: this.algorithm(),
+                iv: this.iv
+            }, this.sessionKey, buffer);
+    }
+
+    sessionKey: CryptoKey;
+    // Cached IV only for testing currently
+    iv: Uint8Array;
+
+    constructor(sessionKey: CryptoKey) {
+        this.sessionKey = sessionKey;
+        // hack for unit testing
+        this.iv = new Uint8Array([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]);
+    }
+
+    algorithm(): string {
+        return "AES-GCM";
+    }
+
+    disposeToWrappedKeyUsing(pkey: PublicKey): PromiseLike<WrappedKey> {
+        try {
+            const webPkey = pkey as WebCryptoPublicKey;
+
+            const rawWrappedKey = crypto.subtle.wrapKey("raw",
+                this.sessionKey,
+                (pkey as WebCryptoPublicKey).cryptoKey(),
+                {   //these are the wrapping key's algorithm options
+                    name: webPkey.algorithm(),
+                }
+            );
+            return rawWrappedKey.then(rawKey => new WebCryptoWrappedKey(new Uint8Array(rawKey), pkey));
+        } finally {
+            // Hopefully this frees the underlying key material
+            this.sessionKey = null;
+        }
+    }
+
+    isDisposed(): boolean {
+        return this.sessionKey != null;
+    }
+
+    fingerprint(): PromiseLike<string> {
+        return Promise.resolve("");
+    }
+}
+
+class WebCryptoDeviceKey extends WebCryptoStorableKey<CryptoKeyPair> implements DeviceKey {
+
+    algorithm(): string {
+        return this.publicKey().algorithm();
+    }
+
+    constructor(key: CryptoKeyPair) {
+        super(key);
+    }
+
+    privateKey(): PrivateKey {
+        return new WebCryptoPrivateKey(this.key.privateKey);
+    }
+
+    publicKey(): PublicKey {
+        return new WebCryptoPublicKey(this.key.publicKey);
+    }
+
+    fingerprint(): PromiseLike<string> {
+        return this.publicKey().fingerprint();
+    }
+}
+
+export class WebCryptoKeyGenerator implements KeyGenerator {
+    generateWrappedStorageKey(deviceKey: DeviceKey): PromiseLike<WrappedKey> {
+        const generatedKey: PromiseLike<CryptoKey> = crypto.subtle.generateKey({name: 'AES-GCM', length: 256},
+            true, ["encrypt", "decrypt", "wrapKey", "unwrapKey"]);
+        return generatedKey.then(key => new WebCryptoSessionKey(key))
+            .then(skey => skey.disposeToWrappedKeyUsing(deviceKey.publicKey()));
+    }
+
+    static getInstance() {
+        return new WebCryptoKeyGenerator();
+    }
+
+    generateAndStoreRecoveryKey(): PromiseLike<RecoveryKey> {
+        return undefined;
+    }
+
+    generateDeviceKey(): PromiseLike<DeviceKey> {
+        const generatedKey: PromiseLike<CryptoKeyPair> = crypto.subtle.generateKey(
```

Comment:
Add note here that we'd like to switch to a different algorithm here (RSA-OAEP is deprecated) but we have to use this for now to get `wrapKey`.